### ISSUE TITLE
Hdfs service should be mandatory for HBase and Hive

### ISSUE THREAD

**Issue by priyank5485** [hortonworks/streamline#630]
If user creates an environment by picking HBase/Hive from an HDP cluster added via ambari then HDFS from same cluster should automatically added by UI to the environment. Reason is HBase has hbase.rootdir that uses hdfs-site.xml core-site.xml to connect to HDFS. Without HDFS being present in environment storm topology fails at runtime throwing an exception from HBaseClient that it cannot connect to hdfs..

Comments from other issue ported here

@shahsank3t Can you or someone familiar with UI look in to this?
@HeartSaVioR I see that we are packaing hdfs-site.xml and core-site.xml with the topology jar if we have just HBase sink in the topology. Just curious about the logic used to pick hdfs-site and core-site if we have two hdfs services(both from different service pools) added to the same environment? Will it pick the right one?

@priyank5485
No we can't determine which is the right one, so the services which rely on configuration file like HDFS, HBase, and Hive should be selected only once.

Maybe better to prevent selecting same service multiple time in an environment until we support multi-clusters for all services properly. That may be an unnecesary limitation for other services, but safest way to avoid that issue.

We might also want to force selecting cluster for each component in order to provide corresponding configurations while deploying.
(Maybe also better to change some fields which are dependent on cluster to be read-only.)

cc. @harshach @shahsank3t

**Comment by sureshms** [comment]
HDFS service definition should be picked up from Hive or HBase and not as HDFS service. We need to understand how hdfs-site.xml is maintained for each of them. They should certainly have their own copy.

**Comment by harshach** [comment]
@priyan5485 @HeartSaVioR 
this will be a difficult problem to solve at this point. Ideally we should just pass required client configs instead of packaging configs but thats for later. 
Even if each service owns their own copy of dependent  service config we can still run into issue if a user selects HDFS from cluster1 and Hive from cluster 2 then we will have two hdfs-site.xml in classpath.
For now lets do minimal change which is to enforce selection of HDFS along with HIVE & HBASE.
We will re-visit this issue post GA. 
@priyank5485 you can file another Issue to take care of UI side of things and lets keep this open for fixing this at the backend.

**Comment by priyank5485** [comment]
Sounds good. Filed this https://github.com/hortonworks/streamline/issues/636
